tests/js/client/server_parameters/detach-threads-cluster.js

```diff
@@ -0,0 +1,177 @@
+  }
+  return {endpointMap, idMap};
+}
+
+function createCollectionWithKnownLeaderAndFollower(cn) {
+  db._create(cn, {numberOfShards:1, replicationFactor:2});
+  // Get dbserver names first:
+  let { endpointMap, idMap } = getEndpointAndIdMap();
+  let plan = arango.GET("/_admin/cluster/shardDistribution").results[cn].Plan;
+  let shard = Object.keys(plan)[0];
+  let coordinator = "Coordinator0001";
+  let leader = plan[shard].leader;
+  let follower = plan[shard].followers[0];
+  return { endpointMap, idMap, coordinator, leader, follower, shard };
+}
+
+function detachSchedulerThreadsSuite2() {
+  'use strict';
+  const cn = 'UnitTestsDetachSchedulerThreads';
+
+  const setupTeardown = function () {
+    db._drop(cn);
+  };
+
+  return {
+    setUp: setupTeardown,
+    tearDown: setupTeardown,
+    
+    testDelayReplicationInExclusiveTrx: function() {
+      // We want to create a lockup in leaseManagedTrx.
+
+      let collInfo = createCollectionWithKnownLeaderAndFollower(cn);
+      // We have a shard whose leader and follower is known to us.
+      
+      let followerEndpoint = collInfo.endpointMap[collInfo.follower];
+
+      // Let's insert some documents:
+      let c = db._collection(cn);
+      let l = [];
+      for (let i = 0; i < 100; ++i) {
+        l.push({_key:"K"+i, Hallo:i});
+      }
+      c.insert(l);
+
+      try {
+        // Block replication in the follower, such that it can be released
+        // later on (nested failure points).
+        debugSetFailAt(followerEndpoint, "synchronousReplication::blockReplication");
+
+        // Create a transaction T writing to the replicated collection
+        let trx = arango.POST_RAW("/_api/transaction/begin", {collections:{write:[cn]}});
+        assertEqual(201, trx.code);
+        trx = trx.parsedBody.result.id;
+
+        // Perform one insert within T in the background, it will block, since
+        // it cannot replicate.
+        let blocked = arango.POST_RAW(`/_api/document/${cn}`, {Hallo:1}, {"x-arango-async": "store", "x-arango-trx-id": trx});
+        assertEqual(202, blocked.code);
+        assertTrue(blocked.headers.hasOwnProperty("x-arango-async-id"));
+        blocked = blocked.headers["x-arango-async-id"];
+
+        // Perform multiple read operations within T, they should fight over
+        // the lease of the transaction and block many threads of the same prio.
+        let l = [blocked];
+        for (let i = 0; i < 100; ++i) {
+          let r = arango.POST_RAW(`/_api/document/${cn}`, {_key:"L"+i}, {"x-arango-async": "store", "x-arango-trx-id": trx});
+          assertEqual(202, r.code);
+          assertTrue(r.headers.hasOwnProperty("x-arango-async-id"));
+          l.push(r.headers["x-arango-async-id"]);
+        }
+
+        // Then perform simple reads to the collection. They should soon be
+        // unblocked, because of detached threads.
+        let startTime = new Date();
+        let guck = c.document("K1");
```

Comment:
`guck` is not used later.
```suggestion
        c.document("K1");
```

Reply:
Removed